### PR TITLE
Update azure sdk and fix breaking changes

### PR DESCRIPTION
**Pull request by oscarkilhed**
Taking a stab at updating the azure SDK to ~~1.0~~ 1.2

### CODE REVIEW

File: aztokenprovider/token_provider.go [hunk 1]
```diff
@@ -164,20 +164,20 @@ func (c *managedIdentityTokenRetriever) GetAccessToken(ctx context.Context, scop
...
+	cloudConf    cloud.Configuration
 	tenantId     string
 	clientId     string
 	clientSecret string
 	credential   azcore.TokenCredential
 }
 
 func (c *clientSecretTokenRetriever) GetCacheKey() string {
-	return fmt.Sprintf("azure|clientsecret|%s|%s|%s|%s", c.authority, c.tenantId, c.clientId, hashSecret(c.clientSecret))
+	return fmt.Sprintf("azure|clientsecret|%s|%s|%s|%s", c.cloudConf, c.tenantId, c.clientId, hashSecret(c.clientSecret))
```
Comment (oscarkilhed) [inline aztokenprovider/token_provider.go:175]: This is probably not working as it should

Comment (andresmgot) [reply aztokenprovider/token_provider.go:175]: yes, you can do just `cloudConf.ActiveDirectoryAuthorityHost` to leave it as it was

File: aztokenprovider/token_provider_test.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func TestAzureTokenProvider_getClientSecretCredential(t *testing.T) {
 		assert.IsType(t, &clientSecretTokenRetriever{}, result)
 		credential := (result).(*clientSecretTokenRetriever)
 
-		assert.Equal(t, "https://login.microsoftonline.com/", credential.authority)
+		assert.Equal(t, "https://login.microsoftonline.com/", credential.cloudConf.ActiveDirectoryAuthorityHost)
```
[the remark this answers is not in the document]
Comment (andresmgot) [reply aztokenprovider/token_provider_test.go:99]: those services configs are empty as far as I can see. I don't think we need to test them?

**Review by andresmgot — COMMENTED** [review]
As far as I can see, this is fine (only one comment about the cache key). Were you able to test this in Grafana? (I am getting some compilation errors)

Any opinions @kostrse ?

**Review by andresmgot — APPROVED** [review]
I was able to test this and it's working properly :+1:

**Comment by kostrse** [comment]
@andresmgot Let me review, do not merge without me.

**Review by kostrse — APPROVED** [review]
Tested, it seems to work fine.

**Pull request merged**